src/widget/flex.rs

```diff
@@ -693,11 +685,13 @@ impl Widget for Flex {
         my_size
     }
 
-    fn paint(&mut self, ctx: &mut PaintCtx, env: &Env) {
+    fn paint(&mut self, ctx: &mut PaintCtx) {
         for child in self.children.iter_mut().filter_map(|x| x.widget_mut()) {
-            child.paint(ctx, env);
+            child.paint(ctx);
         }
 
+        // TODO remove env
+        /*
         // paint the baseline if we're debugging layout
         if env.get(Env::DEBUG_PAINT) && ctx.widget_state.baseline_offset != 0.0 {
```

Comment:
The refactor needs to address these kinds of cases. I'd suggest adding `debug_paint` and `debug_widget_id` bool attributes to GlobalPassCtx and going from there.